docs/decisions/0019-oep-58-atlas-translations-design.rst

```diff
@@ -0,0 +1,423 @@
+    │  │  └── LC_MESSAGES
+    │  │      └── django.po
+    │  ├── de
+    │  │  └── LC_MESSAGES
+    │  │      └── django.po
+    │  ├── en
+    │  │  └── LC_MESSAGES
+    │  │      └── django.po
+    │  └── fr_CA
+    │      └── LC_MESSAGES
+    │          └── django.po
+    ├── drag_and_drop_v2
+    │  ├── ar
+    │  │  └── LC_MESSAGES
+    │  │      └── django.po
+    │  ├── en
+    │  │  └── LC_MESSAGES
+    │  │      └── django.po
+    │  └── fr_CA
+    │      └── LC_MESSAGES
+    │          └── django.po
+    └── edx_proctoring
+        ├── ar
+        │  └── LC_MESSAGES
+        │      └── djangojs.po
+        ├── de
+        │  └── LC_MESSAGES
+        │      └── djangojs.po
+        ├── en
+        │  └── LC_MESSAGES
+        │      ├── djangojs.po
+        │      └── django.po
+        └── fr_CA
+            └── LC_MESSAGES
+                └── djangojs.po
+
+
+
+**Notes:**
+
+- The command above is for demonstration purposes and may not work
+  properly yet.
+- The directory name may change from ``edx-platform-links`` to
+  ``edx-platform-modules`` but this is out of the scope of this proposal and
+  have little to no impact on the rest of the proposal.
+
+Changes to the `openedx-translations repo`_
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `openedx-translations repo`_ organizes the translations by the
+GitHub repository name. However, once an XBlock is installed the repository
+name is no longer known to the ``edx-platform``. Therefore, we provide two
+ways to fetch the XBlock translations:
+
+- By repo name: e.g. `translations/xblock-drag-and-drop-v2 directory`_.
+- By module name: e.g.
+  `translations/edx-platform-links/drag_and_drop_v2 directory`_.
+
+This update is already implemented in the `openedx-translations repo`_ as of
+`edx-platform-links PR #353`_ which includes the details of the changes.
+
+
+BlockI18nService support for ``atlas`` Python translations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``get_python_locale_directory`` will support two modes:
+
+#. When ``ENABLE_ATLAS_TRANSLATIONS`` is disabled, the XBlock translations
+   from the ``pip`` packages will be used such as the
+   ``lib/python3.8/site-packages/drag_and_drop_v2/translations/ar/LC_MESSAGES/text.po``
+   path for the Drag and Drop XBlock.
+
+#. When ``ENABLE_ATLAS_TRANSLATIONS`` is enabled, the atlas translations will
+   be used which is located in the ``edx-platform`` in an the git-ignored
+   ``edx-platform/conf/plugins-locale/drag_and_drop_v2/ar/LC_MESSAGES/text.po``
+   path. This file pulled by the ``pull_plugins_translations`` command.
+
+
+XBlockI18nService support for ``atlas`` JavaScript translations
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``XBlockI18nService`` will provide a new centralized
+``get_javascript_locale_path`` method to get the JavaScript translations
+``django.js`` file.
+
+This function needs to be used by the XBlocks in an opt-in backward-compatible
+manner.
```

Comment:
```suggestion
A ``get_javascript_locale_path`` method will be added to the ``XBlockI18nService`` to provide XBlocks the
appropriate path to ``django.js`` translation files. This will allow XBlocks to utilize legacy packaged translations
or ``atlas`` pulled translations depending on configuration settings.
```